### PR TITLE
[ELASTIC-142] Update Elastic to have a safer update plan

### PR DESCRIPTION
**Pull request by loren**
- Parallel deploy by default
- Serial update by default
- Document that if you're going to turn on xpack (or off), you need to set update to parallel then set it back to serial.

### CODE REVIEW

File: frameworks/elastic/docs/elastic-x-pack.md [hunk 1]
```diff
@@ -12,6 +12,9 @@ enterprise: 'no'
 
 You can toggle this setting at any time. This gives you the option of launching an Elastic cluster without X-Pack and then later enabling it. Or, you can run a cluster with X-Pack enabled to try out the commercial features and, if at the end of the 30-day trial period you don't wish to purchase a license, you can disable it without losing access to your data.
 
+Note that you must set the update strategy to `parallel` when you toggle X-Pack in order to force a full cluster restart.
```
Comment (benclarkwood) [inline frameworks/elastic/docs/elastic-x-pack.md]: IMO this should go first in this section. WDYT, @sascala?

File: frameworks/elastic/docs/limitations.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ The maximum number of deployable nodes is constrained by the DC/OS cluster's res
 
 ## Upgrades and rolling configuration updates do not wait for green status
 
-The `serial` deploy strategy does wait for the cluster to reach green before proceeding to the next node.
+During deployment and upgrades, the `serial` strategy does not wait for the cluster to reach green before proceeding to the next node.
```
Comment (benclarkwood) [inline frameworks/elastic/docs/limitations.md]: Can we be specific and say `Elastic` service? `cluster` is a pretty overloaded term.

File: frameworks/elastic/tests/test_sanity.py [hunk 3]
```diff
@@ -160,6 +159,9 @@ def test_xpack_toggle_with_kibana(default_populated_index):
     doc = config.get_document(config.DEFAULT_INDEX_NAME, config.DEFAULT_INDEX_TYPE, 2, service_name=foldered_name)
     assert doc["_source"]["name"] == "X-Pack"
 
+    # reset upgrade strategy to serial
+    config.update_app(foldered_name, {'UPDATE_STRATEGY': 'serial'}, current_expected_task_count)
```
Comment (benclarkwood) [inline frameworks/elastic/tests/test_sanity.py]: Can you add a "wait for deployment / update / recovery" to be complete here? Don't want next test to get tripped up.

File: frameworks/elastic/docs/elastic-x-pack.md (hunk 1 above)
Comment (sascala) [reply frameworks/elastic/docs/elastic-x-pack.md]: agree. it seems cruel to put it later.

File: frameworks/elastic/docs/limitations.md (hunk 2 above)
Comment (loren) [reply frameworks/elastic/docs/limitations.md]: done

File: frameworks/elastic/tests/test_sanity.py (hunk 3 above)
Comment (loren) [reply frameworks/elastic/tests/test_sanity.py]: This already happens inside `config.update_app` [here](https://github.com/mesosphere/dcos-commons/blob/6f184e7d410023087914a297fdae6162299f2c5e/frameworks/elastic/tests/config.py#L184). And then `pre_test_setup` runs before the next test.

Comment (benclarkwood) [reply frameworks/elastic/tests/test_sanity.py]: 👍

**Pull request merged**